spec/unit/lib/wifi_user/use_cases/check_if_whitelisted_email_spec.rb

```diff
@@ -0,0 +1,40 @@
+describe WifiUser::UseCases::CheckIfWhitelistedEmail do
+  subject { described_class.new(gateway: s3_gateway) }
+  let(:s3_gateway) { double(fetch: '^.*@(aaa\.uk)$') }
+
+  context 'given a whitelisted email' do
+    it 'accepts an address matching the regex' do
+      result = subject.execute('someone@aaa.uk')
+      expect(result).to eq(success: true)
+    end
+
+    it 'accepts an address matching the regex regardless of case' do
+      result = subject.execute('SOMEONE@AAA.UK')
+      expect(result).to eq(success: true)
+    end
+  end
+
+  context 'given a non-whitelisted email' do
+    it 'rejects an address not matching the regex' do
+      result = subject.execute('someone@bbb.uk')
+      expect(result).to eq(success: false)
+    end
+  end
+
+  context 'given an invalid email' do
+    it 'rejects an empty email address' do
+      result = subject.execute('')
+      expect(result).to eq(success: false)
+    end
+
+    it 'rejects a nil address' do
+      result = subject.execute(nil)
+      expect(result).to eq(success: false)
+    end
+
+    it 'respects subdomains' do
```

Comment:
I don't think this test adds anything. The regex above is hard coded to require a `.`, so is just another case of the described class being able to do regex

Reply:
I do however think a context should be added for testing the regex pattern we're using for subdomains, to make sure the class will work with the upcoming subdomain regex

Reply:
True